agent/spec/lib/kontena/workers/node_stats_worker_spec.rb

```diff
@@ -138,12 +129,13 @@
   describe '#publish_node_stats' do
 
     before(:each) do
+      subject.instance_variable_set('@node', node)
```

Comment:
There's actually a bug here... the worker could call `start_publish_loop` -> `publish_node_stats` -> `@node.id` before the `start` -> `observe` -> `@node = ...` happens.